wemake_python_styleguide/visitors/ast/loops.py

```diff
@@ -177,6 +180,35 @@ def _check_infinite_while_loop(self, node: AnyLoop) -> None:
             if not isinstance(evaled, ast.Name) and bool(evaled):
                 self.add_violation(InfiniteWhileLoopViolation(node))
 
+    def _check_await_inside_loop(
+        self, node: AnyLoop | AnyComprehension
+    ) -> None:
+        if isinstance(node, AnyComprehension) and any(
+            generator.is_async for generator in node.generators
+        ):
+            return
+
+        bad_loops = (
+            ast.For
+            | ast.DictComp
+            | ast.GeneratorExp
+            | ast.ListComp
+            | ast.SetComp
+        )
+        await_sub_nodes = walk.get_subnodes_by_type(node, ast.Await)
```

Comment:
Basically, that's why I said that it would be easier to visit `await` nodes first and then find forbidden parents from it. See https://github.com/wemake-services/wemake-python-styleguide/blob/2c4b96e6fa2d456974d8923ebbda3b079fd29c36/wemake_python_styleguide/logic/walk.py#L24-L35

Basically, you would just have an `await` node, pass it into `container = get_closest_parent(node, self._forbidden_await_loops)` and then check that the `container` is not an async comprehension. That's it. Three lines of code.